src/Processors/Streaming/ReplayStreamTransform.cpp

```diff
@@ -16,34 +16,145 @@ namespace Streaming
 
 constexpr Int64 MAX_WAIT_INTERVAL_MS = 500;
 
-ReplayStreamTransform::ReplayStreamTransform(const Block & header, Float32 replay_speed_, Int64 last_sn_)
+ReplayStreamTransform::ReplayStreamTransform(const Block & header, Float32 replay_speed_, Int64 last_sn_, const String & replay_time_col_)
     : ISimpleTransform(header, header, true, ProcessorID::ReplayStreamTransformID)
     , replay_speed(replay_speed_)
     , last_sn(last_sn_)
     , enable_replay(last_sn >= 0)
+    , replay_time_col(replay_time_col_)
 {
-    append_time_index = header.getPositionByName(ProtonConsts::RESERVED_APPEND_TIME);
+    time_index = header.getPositionByName(replay_time_col);
+    auto & type = header.getByPosition(time_index).type;
+    /// user defined replay_time_col must be DateTime64
+    if (replay_time_col != ProtonConsts::RESERVED_APPEND_TIME && !isDateTime64(type))
+        throw Exception(
+            fmt::format("ReplayStreamTransform need datatime64 type column, but got {}", type->getName()), ErrorCodes::LOGICAL_ERROR);
```

Comment:
it's invalid settings `replay_time_column` , please use  `ErrorCodes::INVALID_SETTING_VALUE`